aim/db/infra_model.py

```diff
@@ -164,3 +165,54 @@ class ApicAssignment(model_base.Base, model_base.AttributeMixin):
     aim_aid_id = sa.Column(sa.String(64))
     last_update_timestamp = sa.Column(sa.TIMESTAMP, server_default=func.now(),
                                       onupdate=func.now())
+
+
+class VlanRange(model_base.Base):
+    """DB model for vlan ranges under a NestedParameter."""
+
+    __tablename__ = 'aim_vlan_ranges'
+
+    nested_parameter_cluster_name = sa.Column(
+        sa.String(64), sa.ForeignKey('aim_nested_parameter.cluster_name'),
+        primary_key=True)
+    start = sa.Column(sa.Integer, primary_key=True)
+    end = sa.Column(sa.Integer, nullable=False)
+
+
+class NestedParameter(model_base.Base, model_base.AttributeMixin):
+    __tablename__ = 'aim_nested_parameter'
+
+    cluster_name = model_base.name_column(primary_key=True)
+    domain_name = model_base.name_column()
+    domain_type = sa.Column(sa.String(32), nullable=False)
+    domain_infra_vlan = sa.Column(sa.Integer, nullable=False)
+    domain_service_vlan = sa.Column(sa.Integer, nullable=False)
+    domain_node_vlan = sa.Column(sa.Integer, nullable=False)
+
+    vlan_range_list = orm.relationship(VlanRange,
```

Comment:
Whenever we have relationships that are lists, where the list elements aren't exposed as an AIM resource but are ORM objects, we need to make sure these are optimized or we run into performance problems due to deletion and recreation of DB entries b/c their ORM resources are deleted and recreated. We had to do a similar optimization to the SecurityGroupRuleRemoteIp's in this patch: https://github.com/noironetworks/aci-integration-module/commit/3b1fd190293569f4e82902cf69322374e143a3cc

Reply:
Since this nested parameter thing is only for user configuration and is not part of any openStack or ACI resources so this might be ok for now? Especially we might have to back this out once they got the installer change working.